### PR TITLE
refactor: get rid of `$._raw_js_extension` and `$._raw_ggl_extension`

### LINKED ISSUE #144: Get rid of `$._raw_js_extension` and `$._raw_ggl_extension`
As noted in https://github.com/nkrkv/tree-sitter-rescript/pull/143#issuecomment-1304624487

> Another simplification that can be made:

> There are two rules for extension, `$._raw_js_extension` and `$._raw_ggl_extension`. They are used to apply injections highlights but we can do without them.

> Example for `%re()`:

```scheme
(extension_expression
  (extension_identifier) @_name
  (#eq? @_name "re")
  (expression_statement (string ((string_fragment) @regex))))
```

### PR DESCRIPTION
**Pull request by aspeddro**
Close #144

**Merge after #143**

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -1135,7 +1134,7 @@ module.exports = grammar({
       ))
     )),
 
-    extension_expression: $ => prec('call', seq(
+    extension_expression: $ => prec.right(seq(
```
Comment (aspeddro) [inline grammar.js:1137]: This change is to simplify query injections.

Example:

```res
let _ = %raw(j`1 + 1`)
let _ = %raw(j`1 + 1`)
```

```
(source_file [0, 0] - [2, 0]
  (let_binding [0, 0] - [0, 22]
    (value_identifier [0, 4] - [0, 5])
    (extension_expression [0, 8] - [0, 22]
      (extension_identifier [0, 9] - [0, 12])
      (expression_statement [0, 13] - [0, 21]
        (template_string [0, 13] - [0, 21]
          (template_string_content [0, 15] - [0, 20])))))
  (let_binding [1, 0] - [1, 22]
    (value_identifier [1, 4] - [1, 5])
    (call_expression [1, 8] - [1, 22]
      function: (extension_expression [1, 8] - [1, 12]
        (extension_identifier [1, 9] - [1, 12]))
      arguments: (arguments [1, 12] - [1, 22]
        (template_string [1, 13] - [1, 21]
          (template_string_content [1, 15] - [1, 20]))))))
```

The first line is parsed as `extension_expression` and the second as `call_expression`.

Now, both have the same AST

```
(source_file [0, 0] - [2, 0]
  (let_binding [0, 0] - [0, 22]
    (value_identifier [0, 4] - [0, 5])
    (extension_expression [0, 8] - [0, 22]
      (extension_identifier [0, 9] - [0, 12])
      (expression_statement [0, 13] - [0, 21]
        (template_string [0, 13] - [0, 21]
          (template_string_content [0, 15] - [0, 20])))))
  (let_binding [1, 0] - [1, 22]
    (value_identifier [1, 4] - [1, 5])
    (extension_expression [1, 8] - [1, 22]
      (extension_identifier [1, 9] - [1, 12])
      (expression_statement [1, 13] - [1, 21]
        (template_string [1, 13] - [1, 21]
          (template_string_content [1, 15] - [1, 20]))))))
```

Comment (aspeddro) [reply grammar.js:1137]: This is fixed in #143

**Comment by nkrkv** [comment]
#143 merged

**Comment by nkrkv** [comment]
Perfect, thank you!

Love PRs where deletions overcome additions.

image

**Comment by aspeddro** [comment]
Why the CI failed?

**Comment by aspeddro** [comment]
Humm, `test/highlight/embedded.res` break tree-sitter test

**Comment by nkrkv** [comment]
Oh, haven’t had time to check this out. Thank you for pointing to the problem.

I still think it have to be a manual test because the test framework is not capable of testing embeds. I’ll just move the test file to another directory that will be ignored by the test runner.

**Pull request merged**